### PR TITLE
[ re #293 ] Adding Order-Preserving Embeddings

### PR DESCRIPTION
**Pull request by gallais**
Definition of the partial order and a first set of proofs.

### CODE REVIEW

File: src/Data/List/Relation/OrderPreservingEmbedding.agda [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+------------------------------------------------------------------------
+-- Interaction with List operations
+
+  length-↣-≤ : ∀ {xs ys} → xs ↣ ys → length xs ≤ length ys
+  length-↣-≤ base     = ≤-refl
+  length-↣-≤ (weak p) = ≤-step (length-↣-≤ p)
+  length-↣-≤ (copy p) = s≤s (length-↣-≤ p)
+
+  ++-↣ : ∀ {xs ys us vs} → xs ↣ ys → us ↣ vs → xs ++ us ↣ ys ++ vs
```
Comment (MatthewDaggitt) [inline src/Data/List/Relation/OrderPreservingEmbedding.agda]: This and other below, should all use the standard superscript +/- notation for introduction/elimination forms. Also if they are going to have the `↣` symbol in their name, it should be at the front of the name to match the membership modules (ditto with the length proof above).

File: src/Data/List/Relation/OrderPreservingEmbedding.agda [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+------------------------------------------------------------------------
+-- Type and Basic combinators
+
+module _ {a} {A : Set a} where
+
+  infix 3 _↣_
+  data _↣_ : Rel (List A) a where
+    base : [] ↣ []
+    weak : ∀ {xs y ys} → xs ↣ ys → xs ↣ (y ∷ ys)
+    copy : ∀ {x xs ys} → xs ↣ ys → (x ∷ xs) ↣ (x ∷ ys)
```
[the remark this answers is not in the document]
Comment (MatthewDaggitt) [reply src/Data/List/Relation/OrderPreservingEmbedding.agda]: I agree with @gallais , let's keep uniqueness of representation. I'm agnostic to the names to the constructors, all the proposals seem fine.

**Comment by MatthewDaggitt** [comment]
Looks all pretty good. My only major concern is the location and the potential for confusion with the existing sublist implementation. Given what you say in #292, would you have any objection to me renaming this to `Data.List.Relation.Sublist.Inductive`, mentioning that it is commonly known as an OPE at the top of the file and moving the current sublist implementation to `Data.List.Relation.Sublist.Extensional`?

**Comment by gallais** [comment]
I like `Data.List.Relation.Sublist.Inductive`!

**Comment by MatthewDaggitt** [comment]
Okay updated the PR with the suggestions made here, and added a few more proofs. In the end I went with `base`, `skip`, `keep` for the constructor names as they seem the most meaningful to me. I'll merge it in sometime next week if no-one has any objections...

**Pull request merged**